jsapp/xlform/src/view.mandatorySetting.coffee

```diff
@@ -21,6 +21,13 @@ module.exports = do ->
     render: ->
       reqVal = @getChangedValue()
       template = $($viewTemplates.$$render("row.mandatorySettingSelector", "required_#{@model.cid}", reqVal))
+
```

Comment:
Slightly off topic, but after searching for the cause of the issue I'm wondering how does the Validation Criteria get rendered differently from Error Message? Both are text fields but Error Message's text with quotes is maintained. I'm looking at:
https://github.com/kobotoolbox/kpi/blob/6e0cfa313b5a57145075f4696bc0c19202a83df3/jsapp/xlform/src/view.rowDetail.coffee#L202-L209

Reply:
I did not follow up the code to the root, but I think that the difference is that the error message value is being set in a proper JS way, and the validation criteria (before the fix) was trying to put the value in the string. And I don't see that the value is being forwarded with the creation of the error message textbox, so I'm guessing it is being set somewhere else after rendering